service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java

```diff
@@ -0,0 +1,165 @@
+package uk.nhs.adaptors.gp2gp.uat;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.common.service.TimestampService;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.BloodPressureMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.CodeableConceptCdMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ConditionLinkSetMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.DiaryPlanStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.EhrExtractMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.EncounterComponentsMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.EncounterMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ImmunizationObservationStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.MessageContext;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.NarrativeStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ObservationStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.PertinentInformationObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.StructuredObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.EhrExtractTemplateParameters;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+import uk.nhs.adaptors.gp2gp.utils.ResourceTestFileUtils;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class EhrExtractUATTest {
+    private static final String TEST_FILE_DIRECTORY = "/uat/";
+    private static final String INPUT_DIRECTORY = "input/";
+    private static final String OUTPUT_DIRECTORY = "output/";
+    private static final String INPUT_PATH = TEST_FILE_DIRECTORY + INPUT_DIRECTORY;
+    private static final String OUTPUT_PATH = TEST_FILE_DIRECTORY + OUTPUT_DIRECTORY;
+    private static final String INPUT_JSON_WITH_PATHOLOGY_RECORD = "TC4 - 9465701718_Guerra_inv1only_20210212.json";
```

Comment:
Could we make `TC4 - ` into a constant so it's not repeated throughout the constants? Just a suggestion though & definitely not required

Reply:
I would avoid spaces in filenames

Reply:
Does it look better now? 764984545530ece87996d0738d713f8191c29a69 
Thanks for suggestion!

Reply:
It looks better, but I agree with @Broom94's comment that we should probably avoid spaces in filenames as they can cause weird issues sometimes

Reply:
I agree as well. Here we go: edad4344d51fbe7b44a5d05dc384b0817d43c587